### PR TITLE
hp/set item permissions

### PR DESCRIPTION
**Pull request by matveysiva**
Aims to resolve Set permissions #153
Implemented permissions for book.
Todo: Adjust functionality to work with new items logic (#215)

### CODE REVIEW

**Comment by Yannis-Hofmann** [comment]
The feature works. TODO: write tests (last commit contained some ideas but no reasonable tests yet)

**Review by Yannis-Hofmann — CHANGES_REQUESTED** [review]
Internationalization has to be added for permission level.

**Review by Yannis-Hofmann — DISMISSED** [review]

**Comment by Yannis-Hofmann** [comment]
Important: when this is merged on to main, we have to do a database migration so that a permission is created for all existing items giving the (yet to be implemented?) everybody_group manage rights to all items, so no items will be lost.
Our team will try to prepare this migration tomorrow.

**Comment by bweengener** [comment]
> Important: when this is merged on to main, we have to do a database migration so that a permission is created for all existing items giving the (yet to be implemented?) everybody_group manage rights to all items, so no items will be lost.
> Our team will try to prepare this migration tomorrow.

@Yannis-Hofmann Have a look at my comment in the corresponding Issue for that, please. You will find the migration script there.

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -6,6 +6,7 @@
   <%= csrf_meta_tags %>
   <%= csp_meta_tag %>
 
+
```
Comment (Durborough) [inline app/views/layouts/application.html.erb:9]:
```suggestion
```

I suggest to removing this additional empty line.

**Review by Durborough — APPROVED** [review]
Looks good to me, I just found an additional empty line. This could be removed for code cleanliness. Good job!

**Comment by bweengener** [comment]
I have now fixed Issue #276  and #277  as part of this PR as well. 

**Changelog**:
- Migration for `Everyone`-Group creation
- automatic adding of newly created users in `Everyone`-Group
- Migrations for Default-Memberships of existing accounts (required for `main` deployment)
- Visibilities for `personal_group` were removed from frontend and will be set automatically in backend from now on
- Integration of a migration script, so that already existing items (everyone, can_manage) are assigned 

**The current state on this branch resolves all blockers of permission-related stories.**

Please review as soon as possible to get this integrated into `dev` before deployment.

If anyone here can find the time, then transfer the Native SQL queries to AcrtiveRecord queries using the QueryBuilder.

cc: @Yannis-Hofmann , @WilliamBrandt

**Review by Durborough — APPROVED** [review]
Nicely done, some kinda complex stuff going on here and some comments would be appreciated next time, but the code seems to make sense and is readable.

**Pull request merged**